### PR TITLE
Remove redundant code from towners.cpp

### PR DESCRIPTION
**Pull request by AJenbo**
This changes one bavious. Previously walking away from the dead guy
would interupt the "Your death shall be reveanged", the interuption
seamed un natural and the code didn't seam to have been specifically
made for this case (TownCtrlMsg).

### CODE REVIEW

**Comment by AJenbo** [comment]
This also finishes applying clang-tidy for towner.cpp

File: Source/towners.cpp [hunk 1]
```diff
@@ -420,22 +385,16 @@ void TalkToBarOwner(PlayerStruct &player, TownerStruct &barOwner)
...
-		deadguy._tbtcnt = true;
-		deadguy._tTalkingToPlayer = &player;
 		quests[Q_BUTCHER]._qvar1 = 1;
 		player.PlaySpecificSpeach(8);
-		deadguy._tMsgSaid = true;
 		return;
 	}
 
 	if (quests[Q_BUTCHER]._qactive == QUEST_DONE && quests[Q_BUTCHER]._qvar1 == 1) {
 		quests[Q_BUTCHER]._qvar1 = 1;
```
Comment (galaxyhaxz) [inline Source/towners.cpp:397]: This also seems redundant and can be removed. Checks if `quests[Q_BUTCHER]._qvar1 == 1`, then sets it to 1.

Comment (AJenbo) [reply Source/towners.cpp]: So much more redundant an other jank in the quest logic as well :(

Fixed most of it except for the hellfire portion.

**Review by glebm — DISMISSED** [review]

File: Source/towners.cpp [hunk 2]
```diff
@@ -355,365 +314,296 @@ void TownerTalk(_speech_id message)
...
-	if (!gbIsMultiplayer) {
-		if (player._pLvlVisited[3] && quests[Q_LTBANNER]._qactive != QUEST_NOTAVAIL) {
-			if ((quests[Q_LTBANNER]._qactive == QUEST_INIT || quests[Q_LTBANNER]._qactive == QUEST_ACTIVE) && quests[Q_LTBANNER]._qvar2 == 0 && !barOwner._tMsgSaid) {
+
+	if (quests[Q_LTBANNER]._qactive != QUEST_NOTAVAIL) {
+		if (player._pLvlVisited[3] && quests[Q_LTBANNER]._qactive != QUEST_DONE) {
+			if (quests[Q_LTBANNER]._qvar2 == 0) {
 				quests[Q_LTBANNER]._qvar2 = 1;
 				if (quests[Q_LTBANNER]._qactive == QUEST_INIT) {
-					quests[Q_LTBANNER]._qvar1 = 1;
```
Comment (AJenbo) [inline Source/towners.cpp:395]: Controls map resync, revert

File: Source/towners.cpp [hunk 3]
```diff
@@ -355,365 +314,296 @@ void TownerTalk(_speech_id message)
...
-					quests[Q_PWATER]._qvar1 = 2;
-					healer._tbtcnt = true;
-					healer._tTalkingToPlayer = &player;
-					InitQTextMsg(TEXT_POISON5);
-					SpawnUnique(UITEM_TRING, healer.position.x, healer.position.y + 1);
-					healer._tMsgSaid = true;
-				}
-			}
+	if (quests[Q_PWATER]._qactive != QUEST_NOTAVAIL) {
+		if (player._pLvlVisited[1] && quests[Q_PWATER]._qactive == QUEST_INIT) {
```
Comment (AJenbo) [inline Source/towners.cpp:512]:
```suggestion
		if (player._pLvlVisited[1] || player._pLvlVisited[5]) && quests[Q_PWATER]._qactive == QUEST_INIT) {
```
Hellfire allows you to skip level 1 so keep this.

**Pull request merged**